data/net/minecraft/client/renderer/LevelRenderer.mapping

```diff
@@ -60,127 +60,162 @@ CLASS net/minecraft/client/renderer/LevelRenderer
 		ARG 12 ySpeed
 		ARG 14 zSpeed
 	METHOD addParticleInternal (Lnet/minecraft/core/particles/ParticleOptions;ZDDDDDD)Lnet/minecraft/client/particle/Particle;
-		ARG 1 particleData
-		ARG 2 alwaysRender
+		ARG 1 options
+		ARG 2 ignoreRange
 		ARG 3 x
 		ARG 5 y
 		ARG 7 z
 		ARG 9 xSpeed
 		ARG 11 ySpeed
 		ARG 13 zSpeed
 	METHOD addParticleInternal (Lnet/minecraft/core/particles/ParticleOptions;ZZDDDDDD)Lnet/minecraft/client/particle/Particle;
-		ARG 1 particleData
-		ARG 2 alwaysRender
-		ARG 3 minimizeLevel
+		ARG 1 options
+		ARG 2 ignoreRange
+		ARG 3 decreased
 		ARG 4 x
 		ARG 6 y
 		ARG 8 z
 		ARG 10 xSpeed
 		ARG 12 ySpeed
 		ARG 14 zSpeed
 	METHOD allChanged ()V
-		COMMENT Loads all the renderers and sets up the basic settings usage
+		COMMENT Loads all renderers and sets up the basic options usage.
 	METHOD blockChanged (Lnet/minecraft/world/level/BlockGetter;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/world/level/block/state/BlockState;I)V
 		ARG 1 level
 		ARG 2 pos
 		ARG 3 oldState
 		ARG 4 newState
 		ARG 5 flags
 	METHOD buildClouds (Lcom/mojang/blaze3d/vertex/BufferBuilder;DDDLnet/minecraft/world/phys/Vec3;)V
-		ARG 1 buffer
-		ARG 2 cloudsX
-		ARG 4 cloudsY
-		ARG 6 cloudsZ
-		ARG 8 cloudsColor
+		ARG 1 builder
+		ARG 2 x
+		ARG 4 y
+		ARG 6 z
```

Comment:
I'm pretty sure these are the desired coordinates to build the clouds, or at least x, y, z values to start from. 
If you look in renderClouds they are calculated values, not the camera position.

Reply:
Fair, I was working on the use case in the method itself since both x and z there are used to represent the UV. Looking closer, it seems that the position is determined by `f3`, `f4`, and `f5` which are calculated from `d2`, `d3`, and `d4` whose values are those passed in to build the clouds. However, I will go based on what you have above because of the `prevCloud*` fields, since those are set to a floored value of `d2`, `d3` (after some bit manipulation), and `d4`.